Update publish config for deconst.horse.staging

meker12 opened the pull request:
- Update content.d deploy URL and path to eng handbook source repository
- Remove projects no longer published
- Clean up routes.d template configuration. All private-cloud routes use
  user-guide template

Commit:
Update publish config for deconst.horse.staging

- Delete docs-rpc content.d config to remove master branch map
- Update content.d deploy URL and path to eng handbook source repository
- Clean up routes.d template configuration. All private-cloud routes use
  user-guide template

kallimachos commented on config/content.d/staging.horse.json:
```diff
@@ -1,10 +1,8 @@
   "staging.horse": {
     "content": {
-      "/docs-private-cloud/eng-handbook/": "https://github.com/rackerlabs/docs-rpc/handbook/docs/",
-      "/docs/private-cloud/rpc/master/": "https://github.com/rackerlabs/docs-rpc/master/",
```
@meker12 I think I might be missing something. Don't we want staging to build the master branch?

kallimachos reviewed (commented):
Overall looks good. Thanks for cleaning these files up. :)

meker12 committed:
Correct contentIDBase

Content mapping uses the contentIDBase value in the _deconst.json file for the Sphinx project.  For RPC docs, the ContentIDBase values are based on the branch names (master, v11, v12, v13). 

For the engineering handbook, contentIDBase value is the project name: https://github.com/rackerlabs/docs-rpc/eng-handbook/

For configuration, the contentIDbase must be a unique value per Sphinx project build.  By convention, the API doc projects only build one branch, so we used the project name because that guaranteed a unique ID.  For the RPC docs, we used the branch name in the contentIDBase to make the contentIDBase for each branch build unique.

meker12 committed:
Add missing comma

meker12 committed:
Fix syntax error.

meker12 committed:
Restore template mapping for rcbops release notes

Mistakenly removed the release notes mapping for RPC OpenStack Release Notes from https://github.com/rcbops/rpc-openstack/tree/master/releasenotes/source

meker12 committed:
Fix contentIDBase value for eng-handbook
